CHANGELOG.md

```diff
@@ -7,6 +7,10 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 
+### Fixed
```

Comment:
```suggestion
### Changed
```

Since the changelog introduction description says we follow the https://keepachangelog.com style, whose change definitions are:

>  `Changed` for changes in existing functionality. 
> ...
>  `Fixed` for any bug fixes. 

(I know you use something different for Ruby which is fine, but since this is a shared buildpack I think we should follow the actual definition)

Reply:
(I'm not against switching to a different style for all buildpacks in the future, but we'd need to document it somewhere, update the changelog headers/descriptions and ideally historic changelog entries to match etc first :-) )

Reply:
Short: I a pragmatic path forward might be to acknowlage that this is a large change, even if it's not to an application contracted "feature" and use `Changed` and rev a major. Future UI work doesn't always need to be `Changed`.

Long: While technically this is an observable change from a customer perspective, it wouldn't fall under something documented in the Procfile's Application Contract (if it were formally written like https://github.com/heroku/buildpacks-ruby/blob/main/docs/application_contract.md).

I skimmed the available KAC options before making the update and while "fixed" isn't exactly the correct semantics of what we've done here, `Changed` implies a change to functionality (and major version rev) that I don't feel is 100% warranted here.

Even if "Fixed" is the wrong word, it's closer to the right severity/importance than "Changed" would imply. We could perhaps go off book and go with something like `Style` or `Output` etc. but I feel there's a value to limiting ourselves to avoid "needless creativity." So then we must choose one of them.

A pragmatic path forward might be to acknowlage that this is a large change, even if it's not to an application contracted "feature" and use `Changed` and rev a major (IMHO, they're cheap and plentiful and I don't believe in "pre-1.0" versions). But just because this is a style overhaul. Future UI work doesn't always need to be `Changed.`

Reply:
I think `Changed` is the appropriate category, at least in terms of the style-related changes. The output is an existing feature and modifying how that information is presented is a change to this feature.

> a pragmatic path forward might be to acknowlage that this is a large change

Perhaps the solution to recognize the impact of this change is to write a more elaborate changelog entry? It currently only covers the output style change, but doesn't mention that `Procfile` commands are now also written to the output.

> even if it's not to an application contracted "feature" and use Changed and rev a major.

I don't think we consider the build output part of this buildpack's API? If we do, it would warrant a major version rev increment, regardless of the change definition we pick.

> Even if "Fixed" is the wrong word, it's closer to the right severity/importance than "Changed" would imply

"Fixed" could involve anything from "- Remove trailing whitespace in the build output header" to "- Set command to actual `Procfile` value instead of `echo lulz just testing`". A complete rewrite might not even warrant a mention in the changelog. I don't think the significance/importance of anything in a changelog is implied based on the type of change alone?

That being said: Maybe it'd make sense to also add a separate `Added` section entry that clarifies what was added (beyond the style changes). e.g.:

    ### Added
    
    - The build output now includes the command for each process type. ([#252](https://github.com/heroku/buildpacks-procfile/pull/252))

Reply:
Short: I updated it to `Changed`

Long:
IDK if you were here for the discussion, but there was a big "how do we version buildpacks" discussion and I wrote an internal doc as a very opinonated straw-man to be iterated on in an attempt to sell writing down the differences between accidental an intentional buildpack behavior: https://salesforce.quip.com/BX9pA2TevXGu. Or as a PDF for external users:
[versioning-proposal.pdf](https://github.com/user-attachments/files/18199811/versioning-proposal.pdf). We didn't "adopt" it as a policy, but it's still has some tidbits I still consider useful.

> I don't think the significance/importance of anything in a changelog is implied based on the type of change alone? 

I need artifacts that help make subjective decisions more objective. One of those simplifiying artifacts for me is the Changelog. To me, the changelog is a tool to convey severity of changes and set expectations, the wording "changed", "fixed", "added" is a more useful when used for their severity than their semantics. I.e. a breaking change could be a "fix" to a bug. In that situation it's more important to convey "Something big happened" than it is to convey "well technically it happened due to a fix." It's a lot easier with libraries where there's usually more clearly defined boundries between public and private interfaces.

> I don't think we consider the build output part of this buildpack's API? 

From an application contract perspective it's not. The same inputs provide the same outputs (to the lifecycle API). The experience of the user is changing, but not the experience of the application.

From my original doc I wrote:

> - Maintainers may exercise judgment in elevating severity but not decreasing it.

So I think that's what I'll do here. I think this change is important enough to document, if we document it...i'm getting feedback that it's important that the semantics of wording are important in the changelog. I want to have a strong link between changelog verbs and the versioning number, therefore it is fine to release this as a `Changed` section, which I've done.